From: Nicolas Pitre <nico@cam.org>

This code is currently disabled, which explains why no one was affected.

Signed-off-by: Nicolas Pitre <nico@marvell.com>
---
 arch/arm/lib/memmove.S |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: linux-2.6.26-rc5/arch/arm/lib/memmove.S
===================================================================
--- linux-2.6.26-rc5.orig/arch/arm/lib/memmove.S
+++ linux-2.6.26-rc5/arch/arm/lib/memmove.S
@@ -60,6 +60,7 @@ ENTRY(memmove)
 	CALGN(	bcs	2f			)
 	CALGN(	adr	r4, 6f			)
 	CALGN(	subs	r2, r2, ip		)  @ C is set here
+	CALGN(	rsb	ip, ip, #32		)
 	CALGN(	add	pc, r4, ip		)
 
 	PLD(	pld	[r1, #-4]		)
@@ -139,7 +140,6 @@ ENTRY(memmove)
 		blt	14f
 
 	CALGN(	ands	ip, r1, #31		)
-	CALGN(	rsb	ip, ip, #32		)
 	CALGN(	sbcnes	r4, ip, r2		)  @ C is always set here
 	CALGN(	subcc	r2, r2, ip		)
 	CALGN(	bcc	15f			)

-------------------------------------------------------------------
List admin: http://lists.arm.linux.org.uk/mailman/listinfo/linux-arm-kernel
FAQ:        http://www.arm.linux.org.uk/mailinglists/faq.php
Etiquette:  http://www.arm.linux.org.uk/mailinglists/etiquette.php
